src/amo/components/UserProfileEdit/index.js

```diff
@@ -0,0 +1,364 @@
+/* @flow */
+import * as React from 'react';
+import Textarea from 'react-textarea-autosize';
+import Helmet from 'react-helmet';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+
+import Link from 'amo/components/Link';
+import NotFound from 'amo/components/ErrorPage/NotFound';
+import {
+  editUserAccount,
+  fetchUserAccount,
+  getCurrentUser,
+  getUserByUsername,
+  hasPermission,
+} from 'amo/reducers/users';
+import { USERS_EDIT } from 'core/constants';
+import { withErrorHandler } from 'core/errorHandler';
+import translate from 'core/i18n/translate';
+import log from 'core/logger';
+import { sanitizeHTML } from 'core/utils';
+import Button from 'ui/components/Button';
+import Card from 'ui/components/Card';
+import type { UsersStateType, UserType } from 'amo/reducers/users';
+import type { DispatchFunc } from 'core/types/redux';
+import type { ErrorHandlerType } from 'core/errorHandler';
+import type { I18nType } from 'core/types/i18n';
+
+import './styles.scss';
+
+
+type Props = {|
+  currentUser: UserType | null,
+  dispatch: DispatchFunc,
+  errorHandler: ErrorHandlerType,
+  hasEditPermission: boolean,
+  i18n: I18nType,
+  isEditing: boolean,
+  isEditingCurrentUser: boolean,
+  params: {| username: string |},
```

Comment:
I find this `params` prop a bit odd. Why does this need to take an object that only has one property? I understand that we have another prop for `username`, but maybe this could be `requestedUsername`, or something like that? I don't see the utility in having an object.

Reply:
This prop `params` comes from the react-router.

Reply:
Ok, at least I understand why it's here now. :-)